recipes/packages.rb

```diff
@@ -15,9 +15,26 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+include_recipe 'yum-ius' if node['osl-php']['use_ius']
 
-if node['osl-php']['packages'].any?
-  node.normal['php']['packages'] = node['osl-php']['packages']
+# Get package prefix from version (e.g. "php71u" or "php")
+prefix = if node['osl-php']['use_ius']
+           'php' + node['php']['version'].split('.')[0, 2].join + 'u'
+         else
+           'php'
+         end
+
+php_packages = node['osl-php']['packages']
+
+# Prepend PHP package prefix to short packages (e.g. "php71u-memcached")
+if node['osl-php']['short_packages'].any?
+  php_packages += node['osl-php']['short_packages']
+                  .map { |pkg| prefix + '-' + pkg }
+end
+
+if php_packages.any? || node['osl-php']['use_ius']
+  node.default['php']['packages'] = php_packages
+  node.default['php']['packages'] <<= prefix # Also the name of the main package
```

Comment:
This could be written on one line as `node.default['php']['packages'] = php_packages << prefix`

Also if someone includes this recipe but then specifies no packages, the main package won't be installed. I don't know if this is what you intended.

Reply:
>Also if someone includes this recipe but then specifies no packages and doesn't use use_ius, the main package won't be installed. I don't know if this is what you intended.

I don't think that would be the case. If neither of `osl-php`'s attributes are used to specify packages, and `use_ius` is not set, none of the conditionals would evaluate to true. In that case, all the recipe does is include `php::package`. None of the upstream attributes would have been changed.

This is so that users can choose to use the default upstream packages by not modifying the `node['osl-php']['packages']` and `node['osl-php']['short_packages']` attributes.

Reply:
@doublej472 
>This could be written on one line as node.default['php']['packages'] = php_packages << prefix

For some silly reason this actually causes one of the ChefSpec contexts (the old method) to fail to converge:
```
<Chef::Exceptions::ImmutableAttributeModification: Node attributes are read-only when you do not spe...fy which precedence level to set. To set an attribute use code like `node.default["key"] = "value"'>
```

Not entirely sure why, but I don't know if this is too heinous of a style crime to worry about it.